app/ui/search.tsx

```diff
@@ -0,0 +1,54 @@
+'use client';
+
+import { Suspense, type InputHTMLAttributes } from 'react';
+import { usePathname, useRouter, useSearchParams } from 'next/navigation';
+import { useDebouncedCallback } from 'use-debounce';
+import { MagnifyingGlassIcon } from '@heroicons/react/24/outline';
+
+type Props = {
+  width?: string;
+};
+
+function Search({ width, ...args }: Props & InputHTMLAttributes<HTMLInputElement>) {
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const { replace } = useRouter();
+
+  const handleSearch = useDebouncedCallback((term: string) => {
+    const params = new URLSearchParams(searchParams);
+    if (term) {
+      params.set('query', term);
+    } else {
+      params.delete('query');
+    }
+    replace(`${pathname}?${params.toString()}`);
+  }, 300);
+
+  return (
+    <div className="relative">
+      <MagnifyingGlassIcon className="absolute top-2 left-2 size-6 text-neutral-700 dark:text-neutral-400" />
+      <label htmlFor="search" className="sr-only">
+        Search
+      </label>
+      <input
+        type="search"
+        className={`${
+          width ? width : 'w-full'
+        } h-10 pl-10 pr-2 bg-transparent rounded-md border-none focus:outline-none appearance-none bg-zinc-200 dark:bg-zinc-800 transition placeholder:text-neutral-400`}
+        onChange={(e) => {
+          handleSearch(e.target.value);
+        }}
+        defaultValue={searchParams.get('query')?.toString()}
+        {...args}
+      />
+    </div>
+  );
+}
+
+export default function SearchBar() {
+  return (
+    <Suspense>
```

Comment:
SearchBar에 한정해서 Suspense를 걸어주신 이유가 궁금합니다!

Reply:
PR 본문에 써두었던 것처럼 build 과정에서 에러가 발생해서 해결책으로 걸어두었어요.

> Reading search parameters through useSearchParams() without a Suspense boundary will opt the entire page into client-side rendering. This could cause your page to be blank until the client-side JavaScript has loaded.
https://nextjs.org/docs/messages/missing-suspense-with-csr-bailout